### PR TITLE
Setup Code ql

### LINKED ISSUE #410: Setup CodeQL
> - Turns out its a MetaMask product, which prompted me to check out their CI and looks like they have a ton more scripts to consider! [CodeQL](https://github.com/MetaMask/metamask-extension/actions/workflows/codeql-analysis.yml) looks interesting.

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquity-dollar/discussions/197#discussioncomment-3641178_

### PR DESCRIPTION
**Pull request by FibrinLab**
Resolves #410

### CODE REVIEW

**Comment by FibrinLab** [comment]
Created a CodeQL workflow. Currently only supports Javascript so Solidity files are ignored. Thanks

**Comment by FibrinLab** [comment]
@pavlovcik I noticed CodeQL scanning revealed 8 alerts with 2 high severity security issues. How do we address this?

**Comment by 0x4007** [comment]
> @pavlovcik I noticed CodeQL scanning revealed 8 alerts with 2 high severity security issues. How do we address this?

Post a new issue it'll be a new bounty

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
```
Comment (Steveantor) [inline .github/workflows/codeql-analysis.yml]: CodeQL also supports `typescript`

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Init CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
```
Comment (Steveantor) [inline .github/workflows/codeql-analysis.yml]: CodeQL v1 is being deprecated by the 18th of this month.

**Review by Steveantor — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (FibrinLab) [reply .github/workflows/codeql-analysis.yml]: Added typescript support

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (FibrinLab) [reply .github/workflows/codeql-analysis.yml]: Updated version.

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Init CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
```
Comment (Steveantor) [inline .github/workflows/codeql-analysis.yml]: `github/codeql-action/analyze@v2`
@FibrinLab

Comment (FibrinLab) [reply .github/workflows/codeql-analysis.yml]: Fixed.

**Review by Steveantor — APPROVED** [review]

**Review by Steveantor — APPROVED** [review]

**Review by Steveantor — APPROVED** [review]

**Review by Steveantor — APPROVED** [review]

**Review by Steveantor — APPROVED** [review]

**Review by Steveantor — APPROVED** [review]

**Review by Steveantor — APPROVED** [review]

**Pull request merged**